### ISSUE TITLE
problems with module containing system/bin and lib64

### ISSUE THREAD

**Issue by tombriden** [topjohnwu/Magisk#4]
I'm trying to create a module to install openssh and rsync binaries, this did work in Magisk v6.

the module files are:

```
system/bin/sshd
system/bin/ssh
system/lib64/libssh.so
system/xbin/rsync
system/placeholder
```

When the `lib64/libssh.so` file is included, the device bootloops. It appears that during the dummy mount the real contents of lib64 get lost briefly and it never recovers

```
--------- beginning of crash
01-09 13:42:36.943  5283  5283 F libc    : CANNOT LINK EXECUTABLE "log": file offset for the library "libcutils.so" >= file size: 0 >= 0
01-09 13:42:36.943  5283  5283 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 5283 (log)
01-09 13:42:36.964  5286  5286 F libc    : CANNOT LINK EXECUTABLE "log": file offset for the library "libcutils.so" >= file size: 0 >= 0
01-09 13:42:36.964  5286  5286 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 5286 (log)
01-09 13:42:36.974  5288  5288 F libc    : CANNOT LINK EXECUTABLE "log": file offset for the library "libcutils.so" >= file size: 0 >= 0
01-09 13:42:36.974  5288  5288 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 5288 (log)
01-09 13:42:36.995  5294  5294 F libc    : CANNOT LINK EXECUTABLE "log": file offset for the library "libcutils.so" >= file size: 0 >= 0
01-09 13:42:36.995  5294  5294 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 5294 (log)
```

Trying with just the binaries, I get ssh and sshd in `/system/bin` 

```
1|angler:/ $ ls -l /magisk/.core/dummy/system/bin/ssh*
-rwxr-xr-x 1 root root 182800 2016-10-04 19:59 /magisk/.core/dummy/system/bin/ssh
-rwxr-xr-x 1 root root 245448 2016-10-04 19:59 /magisk/.core/dummy/system/bin/sshd

angler:/ $ ls -l /system/bin/ssh*                   
-rwxr-xr-x 1 root root 182800 2016-10-04 19:59 /system/bin/ssh
-rwxr-xr-x 1 root root 245448 2016-10-04 19:59 /system/bin/sshd
```

but phh's su is no longer available:

```
angler:/ $ echo $PATH
/magisk/.core/bin:/sbin:/vendor/bin:/system/sbin:/system/bin:/magisk/.core/busybox:/system/xbin
angler:/ $ ls -l /magisk/.core/bin
ls: /magisk/.core/bin: No such file or directory
```

here's everything in /magisk (except .core)

```
/magisk/00roothelper/supersu.sh
/magisk/00roothelper/supersu.prop
/magisk/00roothelper/post-fs-data.sh
/magisk/00roothelper/phh.sh
/magisk/00roothelper/phh.prop
/magisk/phh/bin/su
/magisk/phh/bin/sepolicy-inject
/magisk/phh/module.prop
/magisk/phh/post-fs-data.sh
/magisk/angler_openssh/system/bin/sshd
/magisk/angler_openssh/system/bin/ssh
/magisk/angler_openssh/system/xbin/rsync
/magisk/angler_openssh/auto_mount
/magisk/angler_openssh/module.prop
/magisk/angler_openssh/service.sh
```

files in .core

```
1|angler:/ $ ls -l /magisk/.core
total 24
drwxr-xr-x 3 root root 4096 1970-01-09 13:19 dummy
-rw-r--r-- 1 root root   56 2009-01-01 08:00 hosts
drwxr-xr-x 3 root root 4096 1970-01-09 13:19 mirror

```

**Comment by topjohnwu** [comment]
Thanks for the heads up, I remember to leave the essential binaries but forget about the C libraries. Will be fixed in the next release.

**Comment by topjohnwu** [comment]
Fixed in f1d3e35aacdaf8bc0e56246129805077f3431b9d

**Comment by avently** [comment]
Are you sure you fixed it? Because when i try to include a lib to lib64 folder my phone stucks on logo. Without it the phone can boot. Here is a lib 
If i use symlink to /system/lib64/libsodium.so the system also can boot and dnscrypt works (this lib for dnscrypt)

[libsodium.zip](https://github.com/topjohnwu/Magisk/files/640827/libsodium.zip)

**Comment by topjohnwu** [comment]
@avently are you building from master? Or are you referring to v9?

**Comment by avently** [comment]
@topjohnwu i'm reffering to v9

**Comment by laggardkernel** [comment]
@avently I didn't encounter stuck at boot logo when mounting lib64 using v9. Make sure there is **NO binary** files to be mounted to **/bin** in your module. All the boot stucks on v9 are caused by binaries.

The problem about mounting lib64 using v9 i experience, is that some libs are left unmounted when you try to **add new libs and replace existing libs at the same time**. And the problem i mentioned here was fixed by commit 670fe85.
